Null pointer after change orientation

02-25 07:54:26.607 26076-26076/? E/AndroidRuntime: FATAL EXCEPTION: main
                                                   Process: pl.edu.zut.mad.appwizut2, PID: 26076
                                                   java.lang.RuntimeException: Unable to start activity 
This error occured during first run new commits, when I change orientations.


ComponentInfo{pl.edu.zut.mad.appwizut2/pl.edu.zut.mad.appwizut2.activities.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'android.view.View android.view.View.findViewById(int)' on a null object reference
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2416)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
                                                       at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4077)
                                                       at android.app.ActivityThread.-wrap15(ActivityThread.java)
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1350)
                                                       at android.os.Handler.dispatchMessage(Handler.java:102)
                                                       at android.os.Looper.loop(Looper.java:148)
                                                       at android.app.ActivityThread.main(ActivityThread.java:5417)
                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
                                                    Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.view.View android.view.View.findViewById(int)' on a null object reference
                                                       at pl.edu.zut.mad.appwizut2.fragments.FeedFragment.onDataLoaded(FeedFragment.java:79)
                                                       at pl.edu.zut.mad.appwizut2.fragments.FeedFragment.onDataLoaded(FeedFragment.java:27)
                                                       at pl.edu.zut.mad.appwizut2.network.BaseDataLoader.registerAndLoad(BaseDataLoader.java:159)
                                                       at pl.edu.zut.mad.appwizut2.fragments.FeedFragment.onCreateView(FeedFragment.java:55)
                                                       at pl.edu.zut.mad.appwizut2.fragments.PlanChangesFragment.onCreateView(PlanChangesFragment.java:27)
                                                       at android.support.v4.app.Fragment.performCreateView(Fragment.java:1962)
                                                       at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1067)
                                                       at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1248)
                                                       at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1230)
                                                       at android.support.v4.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManager.java:2042)
                                                       at android.support.v4.app.FragmentController.dispatchActivityCreated(FragmentController.java:165)
                                                       at android.support.v4.app.FragmentActivity.onStart(FragmentActivity.java:543)
                                                       at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1237)
                                                       at android.app.Activity.performStart(Activity.java:6268)
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2379)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476) 
                                                       at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4077) 
                                                       at android.app.ActivityThread.-wrap15(ActivityThread.java) 
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1350) 
                                                       at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                       at android.os.Looper.loop(Looper.java:148) 
                                                       at android.app.ActivityThread.main(ActivityThread.java:5417) 
                                                       at java.lang.reflect.Method.invoke(Native Method) 
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726) 
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616) 

--- a/app/src/main/java/pl/edu/zut/mad/appwizut2/fragments/FeedFragment.java
+++ b/app/src/main/java/pl/edu/zut/mad/appwizut2/fragments/FeedFragment.java
@@ -30,6 +30,7 @@ public abstract class FeedFragment extends Fragment implements SwipeRefreshLayou
     private RecyclerView itemListView;
     private SwipeRefreshLayout swipeRefreshLayout;
     private ProgressBar progressBar;
+    private View mNoDataMessageView;
 
 
     private FeedLoader mLoader;
@@ -41,6 +42,7 @@ public abstract class FeedFragment extends Fragment implements SwipeRefreshLayou
         swipeRefreshLayout = (SwipeRefreshLayout)rootView.findViewById(R.id.item_list_swipe_refresh_layout);
         swipeRefreshLayout.setOnRefreshListener(this);
         progressBar = (ProgressBar)rootView.findViewById(R.id.item_list_progress_bar);
+        mNoDataMessageView = rootView.findViewById(R.id.text_no_data);
 
         // TODO: These should be moved to layout xml
         itemListView.setHasFixedSize(true);
@@ -76,12 +78,12 @@ public abstract class FeedFragment extends Fragment implements SwipeRefreshLayou
         swipeRefreshLayout.setRefreshing(false);
 
         if(data != null && data.size() > 0){
-            getView().findViewById(R.id.text_no_data).setVisibility(View.INVISIBLE);
+            mNoDataMessageView.setVisibility(View.INVISIBLE);
             ListItemAdapter listItemAdapter = new ListItemAdapter(data);
             itemListView.setAdapter(listItemAdapter);
         }else {
             itemListView.setAdapter(null);
-            getView().findViewById(R.id.text_no_data).setVisibility(View.VISIBLE);
+            mNoDataMessageView.setVisibility(View.VISIBLE);
         }
     }
 

